computer-use-demo/computer_use_demo/tools/collection.py

```diff
@@ -1,5 +1,6 @@
 """Collection classes for managing multiple tools."""
```

Comment:
adding a retry loop here make sense:
* the tools themselves don't have status codes - they're executed locally against (mostly) shell commands.
* the tool executions are not idempotent/free of side effects; re-executing a failed tool may have unintended consequences